apiserver/facades/agent/provisioner/provisioner.go

```diff
@@ -1414,15 +1425,15 @@ func (a *ProvisionerAPI) CACert() (params.BytesResult, error) {
 
 // CharmProfileChangeInfo retrieves the info necessary to change a charm
 // profile used by a machine.
-func (p *ProvisionerAPI) CharmProfileChangeInfo(machines params.Entities) (params.ProfileChangeResults, error) {
-	results := make([]params.ProfileChangeResult, len(machines.Entities))
+func (p *ProvisionerAPI) CharmProfileChangeInfo(args params.ProfileArgs) (params.ProfileChangeResults, error) {
```

Comment:
I'm wondering if we should keep the old version with the ProvisionerAPIV7

Reply:
Is there a chance we get old agents trying to connect to V7 and how does that work not that we require the application name to upgrade a profile. I guess for V7 we could attempt to work it out potentially?

Reply:
there is a ProvisionerAPIV7. CharmProfileChangeInfo above at 1501.